static/js/message_util.js

```diff
@@ -29,6 +29,14 @@ export function add_old_messages(messages, msg_list) {
     return add_messages(messages, msg_list, {messages_are_new: false});
 }
 
+export function add_new_messages_data(messages, msg_list_data) {
+    if (!msg_list_data.fetch_status.has_found_newest()) {
```

Comment:
This was modeled on `add_new_messages` below, but I’m not sure if the comment there applies here.

Reply:
Yeah, the argument is the same as for the original -- we're trying to maintain a data structure that has a contiguous block of all the messages matching a given filter (in this case, everything including muting), so that we can do things with that data set.  If we add a new message that comes in when the message list is not caught up, it'll corrupt that property, causing incorrect behavior.  (In this case, the use of this data structure for narrowing would result in it thinking it showed you all the messages between IDs X and Y matching that narrow, when in fact it just gave you the messages between X and Y-10000 plus message ID Y).